PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx

```diff
@@ -0,0 +1,623 @@
+
+      // Definition of relativistic momentum and energy to triton and pion and total energy
+      float h3P2 = lnnCand.mom3H[0] * lnnCand.mom3H[0] + lnnCand.mom3H[1] * lnnCand.mom3H[1] + lnnCand.mom3H[2] * lnnCand.mom3H[2];
+      float piP2 = lnnCand.momPi[0] * lnnCand.momPi[0] + lnnCand.momPi[1] * lnnCand.momPi[1] + lnnCand.momPi[2] * lnnCand.momPi[2];
+      float h3E = std::sqrt(h3P2 + h3Mass * h3Mass);
+      float piE = std::sqrt(piP2 + piMass * piMass);
+      float h3lE = h3E + piE;
+
+      // Building the mother particle: lnn
+      std::array<float, 3> lnnMom;
+      const auto& vtx = fitter.getPCACandidate();
+      for (int i = 0; i < 3; i++) {
+        lnnCand.decVtx[i] = vtx[i];
+        lnnMom[i] = lnnCand.mom3H[i] + lnnCand.momPi[i];
+      }
+
+      float lnnPt = std::hypot(lnnMom[0], lnnMom[1]);
+      if (lnnPt < ptMin)
+        continue;
+
+      // Definition of lnn mass
+      float massLNNL = std::sqrt(h3lE * h3lE - lnnMom[0] * lnnMom[0] - lnnMom[1] * lnnMom[1] - lnnMom[2] * lnnMom[2]);
+      bool isLNNMass = false;
+      if (massLNNL > o2::constants::physics::MassTriton - masswidth && massLNNL < o2::constants::physics::MassTriton + masswidth)
+        isLNNMass = true;
+      if (!isLNNMass)
+        continue;
+
+      // V0, primary vertex and poiting angle
+      lnnCand.dcaV0dau = std::sqrt(fitter.getChi2AtPCACandidate());
+      if (lnnCand.dcaV0dau > dcav0dau) {
+        continue;
+      }
+
+      std::array<float, 3> primVtx = {collision.posX(), collision.posY(), collision.posZ()};
+
+      double cosPA = RecoDecay::cpa(primVtx, lnnCand.decVtx, lnnMom);
+      if (cosPA < v0cospa) {
+        continue;
+      }
+
+      for (int i = 0; i < 3; i++) {
+        lnnCand.decVtx[i] = lnnCand.decVtx[i] - primVtx[i];
+      }
+
+      // if survived all selections, propagate decay daughters to PV
+      gpu::gpustd::array<float, 2> dcaInfo;
+
+      o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, posTrackCov, 2.f, fitter.getMatCorrType(), &dcaInfo);
+      lnnCand.isMatter ? lnnCand.h3DCAXY = dcaInfo[0] : lnnCand.piDCAXY = dcaInfo[0];
+
+      o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, negTrackCov, 2.f, fitter.getMatCorrType(), &dcaInfo);
+      lnnCand.isMatter ? lnnCand.piDCAXY = dcaInfo[0] : lnnCand.h3DCAXY = dcaInfo[0];
+
+      // finally, push back the candidate
+      lnnCand.isReco = true;
+      lnnCand.posTrackID = posTrack.globalIndex();
+      lnnCand.negTrackID = negTrack.globalIndex();
+
+      int chargeFactor = -1 + 2 * lnnCand.isMatter;
+      hdEdx3HSel->Fill(chargeFactor * lnnCand.mom3HTPC, h3track.tpcSignal());
+      hNsigma3HSel->Fill(chargeFactor * lnnCand.mom3HTPC, lnnCand.nSigma3H);
+
+      lnnCandidates.push_back(lnnCand);
+    }
+  }
+
+  // Monte Carlo information
+  void fillMCinfo(aod::McTrackLabels const& trackLabels, aod::McParticles const& particlesMC)
```

Comment:
Please check that you only declare parameters that you use (in other functions too).
```suggestion
  void fillMCinfo(aod::McTrackLabels const& trackLabels, aod::McParticles const&)
```